### ISSUE TITLE
Incorrect behaviour of `np.logical_and` with `out` keyword and pandas extension array dtypes from parquet

### ISSUE THREAD

**Issue by mlondschien** [numpy/numpy#19374]
Numpy treats missings in pandas extension arrays as zero when loaded from parquet. See code example:

```python
In [1]: import pandas as pd
   ...: import numpy as np
   ...: 
   ...: df = pd.DataFrame({"I": [0, 1, pd.NA]}, dtype="Int64")
   ...: 
   ...: df.to_parquet("/tmp/file.parquet")
   ...: df_copy = pd.read_parquet("/tmp/file.parquet")
   ...: y = np.ones(3, bool)
   ...: out = np.ones(3, bool)
   ...: 
   ...: np.logical_and(df_copy["I"].values == 0, y, out=out)
Out[1]: 
<BooleanArray>
[True, False, <NA>]
Length: 3, dtype: boolean

In [2]: out
Out[2]: array([ True,  False,  True])

In [3]: np.logical_and(df_copy["I"].values != 0, y, out=out)
Out[3]: 
<BooleanArray>
[False, True, <NA>]
Length: 3, dtype: boolean

In [4]: out
Out[4]: array([False,  True, False])
```

I would expect that `pd.NA == 0` evaluates to `False`.

The same behaviour can be observed for `Float64` and `boolean`:

```python
In [5]: df = pd.DataFrame({
   ...:    "I": pd.array([0, 1, pd.NA], "Int64"),
   ...:    "F": pd.array([0.0, 1.1, pd.NA], "Float64"),
   ...:    "B": pd.array([True, False, pd.NA], "boolean"),
   ...:    "f": np.array([0.0, 1.1, np.nan])
   ...: })
   ...: df.to_parquet("/tmp/file.parquet")
   ...: df_copy = pd.read_parquet("/tmp/file.parquet")
   ...: 
   ...: np.logical_and(df_copy["F"].values == 0, y, out=out)
   ...: out
Out[5]: array([ True, False,  True])

In [6]: np.logical_and(df_copy["B"].values == False, y, out=out)
   ...: out
Out[6]: array([False,  True,  True])
```

For reference, this is the behaviour with `float64`:

```python
In [7]: np.logical_and(df_copy["f"].values == 0, y, out=out)
   ...: out
Out[7]: array([ True, False, False])

In [8]: np.logical_and(df_copy["f"].values != 0, y, out=out)
   ...: out
Out[8]: array([False,  True,  True])
```

Note that all of this only happens if loading from parquet:

```python
In [9]: np.logical_and(df["I"].values == 0, y, out=out)
   ...: out
Out[9]: array([ True, False, False])

In [10]: np.logical_and(df["I"].values != 0, y, out=out)
   ...: out
Out[10]: array([False,  True,  True])
```

@xhochy mentiones that this might be due to missings being stored as zeros in parquet. Related: https://github.com/JDASoftwareGroup/kartothek/issues/484.

### NumPy/Python version information:

Run on `python=3.9.2`, `pandas=1.2.4`, `numpy=1.21.0` and `pyarrow=4.0.1`.

**Comment by xhochy** [comment]
@mlondschien This issue should rather be opened on the `pandas` side. `numpy` is probably only dispatching to `pandas` implementations here.

**Comment by mlondschien** [comment]
Closed in favour of  pandas-dev/pandas#42303.
